app/controllers/api/v1/users_controller.rb

```diff
@@ -16,18 +17,29 @@ def create
     user_response = Faraday.get("https://api.spotify.com/v1/me", nil, header)
     user_params = JSON.parse(user_response.body)
 
-    @user = User.find_or_create_by(
-      username: user_params["display_name"],
-      href: user_params["external_urls"]["spotify"],
-      email: user_params["email"],
-      token: auth_params["access_token"],
-      refresh_token: auth_params["refresh_token"]
-    )
-    session[:user_id] = @user.id
-    redirect_to '/api/v1/dashboard'
+    @user = return_user_by_username(user_params["display_name"])
+
+    if @user == nil
+      @user = User.create(
+        username: user_params["display_name"],
```

Comment:
Way to go! I love how you made it possible to update the token. Getting rid of the find_or_create_by and simply checking the db for the user name is simple and easily understood

Reply:
Super clean, works great onLocalhost